db-migrator-maven-plugin/src/test/java/org/javalite/db_migrator/MergePropertiesSpec.java

```diff
@@ -0,0 +1,126 @@
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.javalite.db_migrator.JdbcPropertiesOverride.*;
+import static org.javalite.test.jspec.JSpec.the;
+
+public class MergePropertiesSpec {
+
+    @Test
+    public void shouldUseEmptyMergePropertiesWithDefaultEnvironment() throws MojoExecutionException {
+        Map<String, Properties> envProperties = execute(null, null);
+        the(compareMap(JSONHelper.toMap("{\n" +
+                "    \"driver\" : \"" + driver() + "\",\n" +
+                "    \"password\" : \"" + password() + "\",\n" +
+                "    \"url\" : \"" + fullUrl() + "\",\n" +
+                "    \"username\" : \"" + user() + "\"\n" +
+                "}"), envProperties.get("null"))).shouldBeTrue();
+    }
+
+    @Test
+    public void shouldUseMergePropertiesWithDefaultEnvironment() throws MojoExecutionException {
+        Map<String, Properties> envProperties = execute(null,"/test_migrations/templator/table-names.properties");
+        the(compareMap(JSONHelper.toMap("{\n" +
+                "    \"authors-table\" : \"authors\",\n" +
+                "    \"books-table\" : \"books\",\n" +
+                "    \"driver\" : \"" + driver() + "\",\n" +
+                "    \"development.authors-table\" : \"development-authors\",\n" +
+                "    \"development.books-table\" : \"development-books\",\n" +
+                "    \"staging.books-table\" : \"staging-books\",\n" +
+                "    \"password\" : \"" + password() + "\",\n" +
+                "    \"url\" : \"" + fullUrl() + "\",\n" +
+                "    \"username\" : \"" + user() + "\"\n" +
+                "}"), envProperties.get("null"))).shouldBeTrue();
+    }
+
+    @Test
+    public void shouldUseMergePropertiesWithDevelopmentEnvironment() throws MojoExecutionException {
+        Map<String, Properties> envProperties = execute("development","/test_migrations/templator/table-names.properties");
+        the(compareMap(JSONHelper.toMap("{\n" +
+                "    \"authors-table\" : \"development-authors\",\n" +
+                "    \"books-table\" : \"development-books\",\n" +
+                "    \"driver\" : \"" + driver() + "\",\n" +
+                "    \"staging.books-table\" : \"staging-books\",\n" +
+                "    \"password\" : \"" + password() + "\",\n" +
+                "    \"url\" : \"" + fullUrl() + "\",\n" +
+                "    \"username\" : \"" + user() + "\"\n" +
+                "}"), envProperties.get("development"))).shouldBeTrue();
+    }
+
+    @Test
+    public void shouldUseMergePropertiesWithDevelopmentAndStagingEnvironments() throws MojoExecutionException {
+        Map<String, Properties> envProperties = execute("development,staging","/test_migrations/templator/table-names.properties");
+        the(compareMap(JSONHelper.toMap("{\n" +
+                "    \"authors-table\" : \"development-authors\",\n" +
```

Comment:
Use multi-line strings here, this is Java 16

Reply:
Replaced by map()